edith-step/src/UserChart.js

```diff
@@ -1,50 +1,64 @@
 import React, {useState, useEffect} from 'react';
-import {Line, Bar, Pie} from 'react-chartjs-2';
+import {Line, Bar, Doughnut} from 'react-chartjs-2';
 import axios from "axios";
+require("regenerator-runtime/runtime")
 
-const LineChart = () => {
-    
-  const [chartData, setChartData] = useState({});
-  const chart = () => {
-    axios
-      .get("/user-stats-servlet")
-      .then(response => {
-        console.log(response);
-        let weekDates = [];
-        let values = [];  
-        const weeklyAggregate = JSON.parse(response.data.weeklyAggregate)
-        weekDates = Object.keys(weeklyAggregate);
-        for (let i = 0; i < weekDates.length; i++) {
-          values.push(weeklyAggregate[weekDates[i]]);
-        }
+async function retrieveWeekData() {
+    const response = await fetch("/user-stats-servlet");
+    const responseJson = await response.json();
+    let weekDates = [];
+    let values = [];  
+    const weeklyAggregate = JSON.parse(responseJson.weeklyAggregate);
+    weeklyAggregate.forEach(week => {
+      weekDates.push(week.date);
+      values.push(week.total);
+    });
+    return [weekDates, values];
+}
+
+const inSameWeek = (itemDate, dateSelection) => {
+  itemDate = new Date(itemDate);
+  dateSelection = new Date(dateSelection);
+  const diffTime = dateSelection - itemDate;
+  const diffDays = diffTime / (1000 * 60 * 60 * 24);
+  return diffDays < 6 && diffDays > 0;
+}
+
+const setChart = (setChartData) => {
+    retrieveWeekData().then(fetchData => {
```

Comment:
nit: parentheses around fetchData for linter

Reply:
Done.